adapters/metax/metax.go

```diff
@@ -0,0 +1,196 @@
+package metax
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/macros"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"github.com/prebid/prebid-server/v2/util/ptrutil"
+)
+
+const SupportedCurrency = "USD"
+
+type adapter struct {
+	template *template.Template
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	// split impressions
+	reqDatas := make([]*adapters.RequestData, 0, len(request.Imp))
+	for i := range request.Imp {
+		imp := &request.Imp[i]
+		impCopy := *imp
+		requestCopy := *request
+
+		metaxExt, err := parseBidderExt(imp)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if err := preprocessImp(&impCopy); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		endpoint, err := a.getEndpoint(metaxExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		requestCopy.Imp = []openrtb2.Imp{impCopy}
+		reqJSON, err := json.Marshal(requestCopy)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+
+		headers := http.Header{}
+		headers.Add("Content-Type", "application/json;charset=utf-8")
+		headers.Add("Accept", "application/json")
+		reqDatas = append(reqDatas, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     endpoint,
+			Body:    reqJSON,
+			Headers: headers,
+			ImpIDs:  openrtb_ext.GetImpIDs(requestCopy.Imp),
+		})
+	}
+
+	return reqDatas, errs
```

Comment:
i cant see where You check req args ("publisherId", "adunit")

Reply:
them are integer in new commit. I think we needn't to check them. do you have some advices?

Reply:
we have set the params checking in bidder-params/metax.json